src/test-table.md

```diff
@@ -100,6 +104,75 @@ for _, tt := range tests {
 }
 ```
 
+### Avoid Messy Table Tests
+
+Table tests become difficult to read and maintain if there becomes significant
+logic forks within the subtest `for` loop. Table tests should **NOT** be used whenever 
+there needs to be complex logic added inside this execution loop.
+
+Large complex table tests harm readability and maintainability because test readers will
+have difficulty debugging test failures that occur.
+
+Table tests like this should be split into either multiple test tables or multiple 
+individual `Test...` functions
+
+Some general guidelines to follow are:
+
+* Tests should focus on the narrowest unit of behavior possible
+* Tests should contain no `if` statements/logic if at all possible
+* All table fields should be used in all table tests
+* All test logic should be evaluated for all table cases
+
+Simple logic, such as `shouldErr` and `expectErr`, that does not significantly impact readability,
+is acceptable as long as the tests are clear and understandable.
+
+A common clear pattern to avoid is the addition of `if` statements for fields such as mock expectations
+and expected behavior that test too much responsibility for a single test.
```

Comment:
We can connect this to the previous list of Do's if we put the "avoid" case first:

```suggestion
Concretely, that means that you should avoid
having multiple branching pathways (e.g. `shouldError`, `expectCall`, etc.),
addition of `if` statements for specific mock expectations (e.g. `shouldCallFoo`),
or placing functions inside the table (e.g. `setupMocks func(*FooMock)`).

As an exception, if the test body is short and straightforward,
it's acceptable to have a single branching pathway for success versus failure cases
with a table field like `wantErr` to specify error expectations.
```